lighthouse-core/audits/seo/hreflang.js

```diff
@@ -32,24 +34,17 @@ const UIStrings = {
 const str_ = i18n.createMessageInstanceIdFn(__filename, UIStrings);
 
 /**
- * Import list of valid languages from axe core without including whole axe-core package
+ * Import list of valid languages from axe core.
  * This is a huge array of language codes that can be stored more efficiently if we will need to
  * shrink the bundle size.
  * @return {Array<string>}
  */
 function importValidLangs() {
-  // @ts-expect-error - global switcheroo to load axe valid-langs
-  const axeCache = global.axe;
-  // @ts-expect-error
-  global.axe = {utils: {}};
+  // Define a window-ish object so axe will export to it.
+  const window = {getComputedStyle: () => {}};
+  eval(axeLibSource);
```

Comment:
axe 4 went to esmodules and https://github.com/dequelabs/axe-core/pull/2527 brought the `valid-langs` module down to 2KiB.

`import()` to pull in the module directly the right way seems simpler than evalling a script meant to run in browsers and a lot more future proof :)

`audit()` is always `await`ed and `VALID_LANGS` is only ever used in `isExpectedLanguageCode()`, so the changes needed to do the `async` conversion seem like they'd be pretty small.

Reply:
I never would have guessed the complexity here lies with converting to async :)

I thought the following two problems stand in the way of esmodules:

- requires experimental flag to run in node 12
- bundling support

Are neither of these issues?

Reply:
> requires experimental flag to run in node 12

Ugh, well it doesn't in node 12, but that [happened in 12.17](https://github.com/nodejs/node/blob/master/doc/changelogs/CHANGELOG_V12.md#2020-05-26-version-12170-erbium-lts-targos) :(

Since we barely agreed to 12.13, I guess we can fix it in LH8.

Reply:
FYI these changes are from #11442